batch-jberet/src/main/java/org/wildfly/extension/batch/jberet/deployment/BatchJobExecutionResource.java

```diff
@@ -52,6 +52,19 @@
     // Should be guarded by it's instance
     private final Set<String> children = new LinkedHashSet<>();
 
+    /**
+     * Last time when job names were refreshed
+     */
+    private long lastRefreshedTime;
```

Comment:
Maybe not crucial, but it might be good to make this `volatile`.

Reply:
I've seen the metrics collector `org.wildfly.extension.metrics.MetricCollector` triggers this refreshing multiple times during a deployment. Sometimes 2, I've also seen 3 times. Every time the stack trace is the same (see the comment in https://issues.redhat.com/browse/WFLY-14275).